rename.g

```diff
@@ -0,0 +1,324 @@
+#############################################################################
+## 
+## This file is part of recog, a package for the GAP computer algebra system
+## which provides a collection of methods for the constructive recognition
+## of groups.
+## 
+## This files's authors include Sergio Siccha, Friedrich Rober
+## Work on this file started in Summer School Matrix Group Recognition 2019
+## 
+## Copyright of recog belongs to its developers whose names are too numerous
+## to list here. Please refer to the COPYRIGHT file for details.
+## 
+## SPDX-License-Identifier: GPL-3.0-or-later
+## 
+## --------------------------------------------------------------------------
+## DESCRIPTION:
+## 
+## This file renames functions of the recog package and generates 
+## a table of renamed functions for the documentation.
+## The renamings_to_execute.csv file specifies how to rename.
+## The renamings_history.csv file stores the history of renamings
+## and is used for the generation of the renaming table in the documentation.
+## --------------------------------------------------------------------------
+## INSTRUCTIONS:
+## 
+## Run it with GAP from within the package directory.
+## There are two possible applications for running this file.
+## --------------------------------------------------------------------------
+## APPLICATIONS:
+## 
+## The first one adresses the developers of the recog package.
+## They can use the renamings_to_execute.csv file in order to
+## discuss new names for functions and afterwards to apply the name changes
+## in the recog package by running this file as instructed.
+## Below an example of this application is included.
+## 
+## The second one adresses people who use the recog package as a dependency.
+## They can run this file with an empty renamings_to_execute.csv file
+## to apply all name changes that are stored in the renamings_history.csv 
+## file to their own projects.
+## --------------------------------------------------------------------------
+## DETAILS:
+## 
+## This file reads the renamings_to_execute.csv file that specifies how to
+## rename. If it is empty, this file attempts to do the renamings specified
+## in the renamings_history.csv file. 
+## The further explanations apply if the file is not empty.
+## 
+## First this file checks if the renamings_to_execute.csv file is correct.
+## The renamings_to_execute.csv file has the following form:
+##     - First column 'OLD NAME' contains the old name.
+##     - Second column 'NEW NAME' contains the new name that is used now.
+##     - Third column 'TYPE' contains the type used for Ref in GAPDoc.
+## Then further checks are done and on failure Errors/Warnings are printed.
+## Primarily they are done to ensure that the renamings_history.csv file 
+## always has duplicate-free columns after merging. 
```

Comment:
With merging I mean the generation of the `renamings_history.csv` after concatenation with the `renamings_to_execute.csv`.
With duplicate-free I wanted to say that there cannot appear an entry `X `twice in a `Old/New Name` column. Maybe this is a little bit too restrictive and we should allow that. Then I would need to change the implementation of the simplification of the csv, where we reduce `X_1 -> ... -> X`_n to `X_1 -> X_n`. Nevertheless I would say that duplicates should still throw a Warning instead of an Error in most cases. We can discuss that for each case individually, if you wish.

Reply:
> Nevertheless I would say that duplicates should still throw a Warning instead of an Error in most cases.

Jepp. I actually think a warning is preferable in all cases, at least in the "duplicates" cases.

Reply:
> Then I would need to change the implementation of the simplification of the csv, where we reduce X_1 -> ... -> X_n to X_1 -> X_n.

That's about "use case 2" though, right? Then I'd suggest let's ignore the possibility of duplicates there (in the simplification function) for now. We can still take care of duplicates _if_ they really appear. After all if a PR introduces a duplicate which doesn't make any sense, that PR shouldn't make it through the reviewing process.